Tests/ConfidenceTests/ConfidenceTest.swift

```diff
@@ -142,14 +142,8 @@ class ConfidenceTest: XCTestCase {
                 resolveReason: .match)
         ]
 
-        let expectation = expectation(description: "context is synced")
-        let cancellable = confidence.contextReconciliatedChanges.sink { _ in
-            expectation.fulfill()
-        }
-        confidence.putContext(context: ["targeting_key": .init(string: "user2")])
-        await fulfillment(of: [expectation], timeout: 1)
-        cancellable.cancel()
 
+        await confidence.putContext(context: ["targeting_key": .init(string: "user2")])
```

Comment:
This test change reflects well the change: users can now wait for the context to be reconciled in the flags' cache, before `getEvaluation`. Before this was not possible, as `contextReconciliatedChanges` is internal